nannyml_cloud_sdk/experiment/experiment.py

```diff
@@ -0,0 +1,304 @@
+        name: User-defined name of the experiment.
+        createdAt: Timestamp when the experiment was created.
+    """
+    id: str
+    name: str
+    experimentType: ExperimentType
+    createdAt: datetime.datetime
+
+
+class ExperimentDetails(ExperimentSummary):
+    """Detailed information about an experiment.
+
+    Attributes:
+        latestRun: The currently active run or latest run performed for the experiment.
+            This is ``None`` if no runs have been performed yet.
+    """
+
+    latestRun: Optional[RunSummary]
+
+
+_EXPERIMENT_SUMMARY_FRAGMENT = f"""
+    fragment ExperimentSummary on Experiment {{
+        {' '.join(ExperimentSummary.__required_keys__)}
+    }}
+"""
+
+_EXPERIMENT_DETAILS_FRAGMENT = """
+    fragment ExperimentDetails on Experiment {
+        ...ExperimentSummary
+        latestRun {
+            ...RunSummary
+        }
+    }
+""" + _EXPERIMENT_SUMMARY_FRAGMENT + RUN_SUMMARY_FRAGMENT
+
+_LIST_EXPERIMENTS = gql("""
+    query listExperiments($filter: ExperimentsFilter) {
+        experiments(filter: $filter) {
+            ...ExperimentSummary
+        }
+    }
+""" + _EXPERIMENT_SUMMARY_FRAGMENT)
+
+_READ_EXPERIMENT = gql("""
+    query readExperiment($id: Int!) {
+        experiment(id: $id) {
+            ...ExperimentDetails
+        }
+    }
+""" + _EXPERIMENT_DETAILS_FRAGMENT)
+
+_GET_EXPERIMENT_DATA_SOURCES = gql("""
+    query getExperimentDataSources($experimentId: Int!) {
+        experiment(id: $experimentId) {
+            dataSource{
+              ...DataSourceSummary
+            }
+        }
+    }
+""" + DATA_SOURCE_SUMMARY_FRAGMENT)
+
+_GET_EXPERIMENT_DATA_HISTORY = gql("""
+    query getModelDataHistory($experimentId: Int!) {
+        experiment(id: $experimentId) {
+            dataSource{
+                events {
+                    ...DataSourceEvent
+                }
+                ...DataSourceSummary
+            }
+        }
+    }
+""" + DATA_SOURCE_SUMMARY_FRAGMENT + DATA_SOURCE_EVENT_FRAGMENT)
+
+
+_CREATE_EXPERIMENT = gql("""
+    mutation createExperiment($input: CreateExperimentInput!) {
+        create_experiment(input: $input) {
+            ...ExperimentDetails
+        }
+    }
+""" + _EXPERIMENT_DETAILS_FRAGMENT)
+
+_DELETE_EXPERIMENT = gql("""
+    mutation deleteExperiment($id: Int!) {
+        delete_experiment(id: $id) {
+            id
+        }
+    }
+""")
+
+_ADD_DATA_TO_DATA_SOURCE = gql("""
+    mutation addDataToDataSource($input: DataSourceDataInput!) {
+        add_data_to_data_source(input: $input) {
+            id
+        }
+    }
+""")
+
+_UPSERT_DATA_IN_DATA_SOURCE = gql("""
+    mutation updateDataInDataSource($input: DataSourceDataInput!) {
+        upsert_data_in_data_source(input: $input) {
+            id
+        }
+    }
+""")
+
+_REMOVE_DATA_FROM_DATA_SOURCE = gql("""
+    mutation removeDataFromDataSource($input: DataSourceDeleteInput!) {
+        delete_data_from_data_source(input: $input) {
+            id
+        }
+    }
+""")
```

Comment:
These mutations seem identical across module. We might want to have these mutations in some form of common file and use them from the different modules. What do you think?

Reply:
Yeah, didn't want to overcomplicate / refactor everything in the beginning, but I ended up splitting stuff up anyway. Forgot about these ones. 

Added them to the `data` module